lib/Interpreter/CIFactory.cpp

```diff
@@ -807,10 +807,20 @@ namespace {
     assert(!strcmp(DE->getName(), cwdbuf) && "Unexpected name for $PWD");
     // Build the virtual file
     const char* Filename = "InteractiveInputLineIncluder.h";
-    const std::string& CGOptsMainFileName
-      = CI->getInvocation().getCodeGenOpts().MainFileName;
-    if (!CGOptsMainFileName.empty())
-      Filename = CGOptsMainFileName.c_str();
+    /*
+     ###FIXME Filename used to be overriden as below..
+        const std::string& CGOptsMainFileName
+          = CI->getInvocation().getCodeGenOpts().MainFileName;
+        if (!CGOptsMainFileName.empty())
+          Filename = CGOptsMainFileName.c_str();
+
+     However when running cling on the command-line any attempt to execute
+     a file in the current working directory fails when this is done.
+     If there'se a chance clang is parsing in parallel while we're initing
+     if would be good to put this back in and figure out how to get the decls
+     in IncrementalParser::ParseInternal
+    */
```

Comment:
I am not sure I understand the comment. Could you add a test case, showing what is the fix about?